presto-common/src/main/java/com/facebook/presto/common/type/UserDefinedType.java

```diff
@@ -19,8 +19,8 @@
 
 public class UserDefinedType
 {
-    private final QualifiedObjectName name;
-    private final TypeSignature representation;
+    protected final QualifiedObjectName name;
+    protected final TypeSignature representation;
```

Comment:
lol, I guess you were thinking extending this class. I would just merge `UserDefinedDistinctType` into this